core/src/main/scala/org/apache/spark/storage/BlockManagerMaster.scala

```diff
@@ -208,8 +219,13 @@ class BlockManagerMaster(
         log"${MDC(ERROR, e.getMessage)}", e)
     )(ThreadUtils.sameThread)
     if (blocking) {
-      // the underlying Futures will timeout anyway, so it's safe to use infinite timeout here
-      RpcUtils.INFINITE_TIMEOUT.awaitResult(future)
+      if (cleanBlockBlockingTimeout.isDefined) {
+        new RpcTimeout(FiniteDuration(cleanBlockBlockingTimeout.get, TimeUnit.SECONDS),
+          CLEANER_REFERENCE_TRACKING_BLOCKING_TIMEOUT.key)
+      } else {
+        // the underlying Futures will timeout anyway, so it's safe to use infinite timeout here
+        RpcUtils.INFINITE_TIMEOUT.awaitResult(future)
+      }
```

Comment:
Could you extract this block into a common function?

Reply:
Could you also add a unit test for the change? For example, a unit test for the common function is ok if it's hard to add end-2-end test.

Reply:
Done